tests/benchmark-runner-tests.mjs

```diff
@@ -106,22 +108,31 @@ describe("BenchmarkRunner", () => {
     });
 
     describe("Suite", () => {
-        describe("_runAllSuites", () => {
-            let _runSuiteStub, _finalizeStub, _removeFrameStub;
+        describe("runAllSuites", () => {
+            let _runSuiteStub, _finalizeStub, _appendFrameStub, _removeFrameStub;
 
             before(async () => {
-                _runSuiteStub = stub(runner, "_runSuite").callsFake(() => null);
-                _finalizeStub = stub(runner, "_finalize").callsFake(() => null);
+                _runSuiteStub = stub(runner, "runSuite").callsFake(async () => null);
+                _finalizeStub = stub(runner, "_finalize").callsFake(async () => null);
+                _appendFrameStub = stub(runner, "_appendFrame").callsFake(async () => null);
                 _removeFrameStub = stub(runner, "_removeFrame").callsFake(() => null);
+                for (const suite in runner.suites)
+                    spy(suite, "prepare");
 
-                await runner._runAllSuites();
+                await runner.runAllSuites();
+            });
+
+            it("should call prepare on all suites", () => {
+                for (const suite in runner.suites)
+                    suite.prepare.calledOnce();
```

Comment:
I'm surprised this is working. I think `calledOnce` is a property on a spy, not a function?
You probably wanted `assert.calledOnce(suite.prepare)`. This makes me wonder if we really execute this line.

Debugging further it looks like indeed `runner.suites` is undefined, that's why the test passes (the for loop never runs anything). For loops are dangerous in tests :-)
I think you wanted `_suites`, not `suites` (also above in the `before`). I tried this but then I get more errors, so I'll let you debug :-D

Also it would be good to add something such as `expect(runner._suites).not.to.have.length(0)` to make sure we enter the loop.

Reply:
good catch, added some more checks.